sklearn/neighbors/classification.py

```diff
@@ -383,10 +408,111 @@ def predict(self, X):
 
             y_pred[inliers, k] = classes_k.take(mode)
 
-        if outliers:
-            y_pred[outliers, :] = self.outlier_label
+            if outliers:
+                if self.outlier_label == 'most_frequent':
+                    prior = np.bincount(_y[:, k])
+                    frequent_label_index = prior.argmax()
+                    y_pred[outliers, k] = classes_k[frequent_label_index]
 
+        if outliers and isinstance(self.outlier_label, (numbers.Integral,
+                                                        np.integer)):
+            y_pred[outliers, :] = self.outlier_label
+            warns.warn('No neighbors found for test samples %r, '
```

Comment:
I'm not sure why eye now want to warn. The user has asked us to handle outliers in a particular way, so why should we be noisy about it.